### DISCUSSION TITLE
`kitty` stress test: truecolor 60fps video working @ 720p ; 30fps @ 1280p -- via `tpix`!!

### DISCUSSION THREAD

**Discussion by ctrlcctrlv** [kovidgoyal/kitty#5660] [Show and tell]
<a href="https://www.youtube.com/watch?v=-Xix6FYe63M"><img src="https://i3.ytimg.com/vi/-Xix6FYe63M/maxresdefault.jpg"></a>

It's certain more performance is possible. I did it with `tpix` because the default `kitty +icat` is too slow because it's written in Python. But that could be remedied if it were to have a mode wherein it loops on a file, continuously attempting to read that file, and then outputs whatever is in it. Or, the graphics protocol itself could use the kernel's ability to watch files for writes, and whenever it sees one, update the file. Or other possibilities.

# `pty1`
```bash
ffmpeg -re -hwaccel nvdec -i rickroll.mkv -vf "hwupload,scale_npp=720:-1,hwdownload,format=nv12" -autoscale 0 \
    -vsync 1 -f image2 -c:v bmp -atomic_writing 1 -update 1 -r 60 -y /tmp/lol.bmp -f pulse -prebuf 30000 hw,0
```

# `pty2`
```bash
while true; do tpix -n /tmp/lol.bmp; done; reset
```

**Comment by kovidgoyal** [comment]
Cool, just so you know, I plan to rewrite icat as a statically compiled
executable using Go, (see the go branch). At which time it will use shared memory
to transfer pixels which should greatly improve performance apart from
being written in a faster language.

**Comment by bee-on-the-green** [comment]
Thank you very much ctrlcctrlv!

I am building a little conky for my desktop (it's not a conky, a terminal window rather). So far, it can only generate random quotes every couple of seconds (2 words greped from the dictionary; linked by a 'is' or 'are', depending whether the first word is detected as being singular or plural).

I would like to include pictures/videos, also randomly selected from my home folder (not long ago, I did not know it was possible to display images in a terminal).

Even thought my knowledge is quite poor (I only know beginner level Bash), I am going to try to include some tpix's feature in my script. Hopefuly, I will manage to include some features in my script. That would be a start.

The idea of having something fast is quite interesting, as it takes my computer more than a second just to generate a quote! (probably because my script is poorly designed). 

At any rate, thank you for providing ideas to users as well as to people working on this project!

![image](https://github.com/kovidgoyal/kitty/assets/118056528/18af6c0c-11de-440a-8c3f-f97433fa64ad)
